ui-admin/src/study/surveys/SurveyEditorView.tsx

```diff
@@ -21,24 +24,66 @@ const SurveyEditorView = (props: SurveyEditorViewProps) => {
     onSave
   } = props
 
-  const [editedContent, setEditedContent] = useState<string>()
+  const FORM_DRAFT_KEY = getFormDraftKey({ form: currentForm })
+
   const [isEditorValid, setIsEditorValid] = useState(true)
   const [saving, setSaving] = useState(false)
-  const isSaveEnabled = !!editedContent && isEditorValid && !saving
+  const [savingDraft, setSavingDraft] = useState(false)
+
+  //Let the user know if we loaded a draft from local storage when the component first renders. It's important
+  //for them to know if the version of the survey they're seeing are from a draft or are actually published.
+  const [showLoadedDraftModal, setShowLoadedDraftModal] = useState(!!getDraft({ formDraftKey: FORM_DRAFT_KEY }))
+  const [showDiscardDraftModal, setShowDiscardDraftModal] = useState(false)
+
+  const [draft, setDraft] = useState<FormDraft | undefined>(
+    !readOnly ? getDraft({ formDraftKey: FORM_DRAFT_KEY }) : undefined)
+
+  const isSaveEnabled = !!draft && isEditorValid && !saving
+
+  const draftRef = useRef<FormDraft | undefined>()
```

Comment:
having both draft as a state and a ref is confusing. I think you'll be able to avoid it if you follow the same patterns that we use for auto-saving survey responses for participants.   you'll need to use a somewhat non-intuitive mode of React's useState hooks where you call setDraft with a function.  Then React will always pass that function the latest value of the state thing, so you don't have to use a ref.  This is counter-intutitive because you'll be calling setDraft even when you don't want to change the value, but it's actually the best practice. 
```
  setDraft(currentDraft => {
    // currentDraft is guaranteed to be up-to-date 
    // do some stuff
    return currentDraft
})
```

Reply:
yeah, that's the idea -- you can't pass the draft itself because the callback will capture it and then get stale, but you can pass the setDraft function.  The set[X] function is guaranteed to always return the latest version whenever it is called, and set[X] itself doesn't change between renders

Reply:
Oh, I think I see. You're suggesting using the state setter to get the latest state value. Sneaky.

Reply:
Ah TIL, that's an interesting way to do this. I took a shot at it in c11751e893365e1b22b13acc316deb2955753cd5

Also handles draft comparison to avoid useless autosaves